__tests__/nc_news_db.test.js

```diff
@@ -37,6 +37,128 @@ describe("GET /api/", () => {
+					expect(articles.length).not.toBe(0);
+					expect(articles).toBeSortedBy("title");
+				});
+		});
+
+		it("status: 200, returns articles sorted by author in descending order", () => {
+			return request(app)
+				.get("/api/articles?sorted_by=author&order=desc")
+				.expect(200)
+				.then(({ body }) => {
+					const articles = body;
+					expect(articles).toBeSortedBy("author", { descending: true });
+				});
+		});
+
+		it("status: 200, returns articles sorted by votes in ascending order", () => {
+			return request(app)
+				.get("/api/articles?sorted_by=votes&order=asc")
+				.expect(200)
+				.then(({ body }) => {
+					const articles = body;
+					expect(articles).toBeSortedBy("votes");
+				});
+		});
+
+		it("status: 200, returns articles sorted by topic in descending order", () => {
+			return request(app)
+				.get("/api/articles?sorted_by=topic&order=desc")
+				.expect(200)
+				.then(({ body }) => {
+					const articles = body;
+					expect(articles).toBeSortedBy("topic", { descending: true });
+				});
+		});
+
+		it("status: 400, returns an error when given a invalid sort_by column", () => {
+			return request(app)
+				.get("/api/articles?sorted_by=not_a_column")
+				.expect(400)
+				.then(({ body }) => {
+					const articles = body;
+					expect(articles.message).toBe("Invalid query parameter");
+				});
+		});
+
+		it("status: 400,returns an error when given an invalid order value", () => {
+			return request(app)
+				.get("/api/articles?order=bottom")
+				.expect(400)
+				.then(({ body }) => {
+					const articles = body;
+					expect(articles.message).toBe("Invalid query parameter");
+				});
+		});
+
+		it("status: 400,returns an error when given a non-string sort_by value", () => {
+			return request(app)
+				.get("/api/articles?sorted_by=8080")
+				.expect(400)
+				.then(({ body }) => {
+					const articles = body;
+					expect(articles.message).toBe("Invalid query parameter");
+				});
+		});
+
+		it("status: 400, returns an error when given a non-string order value", () => {
+			return request(app)
+				.get("/api/articles?order=8080")
+				.expect(400)
+				.then(({ body }) => {
+					const articles = body;
+					expect(articles.message).toBe("Invalid query parameter");
+				});
+		});
+	});
+
+	describe("GET /api/articles (topic queries)", () => {
+		it("status: 200, should filter articles by an existing topic", () => {
+			return request(app)
+				.get("/api/articles?topic=mitch")
+				.expect(200)
+				.then(({ body }) => {
+					const articleArray = body;
+					expect(articleArray.length).not.toBe(0);
+					articleArray.forEach((article) => {
+						expect(article.topic).toBe("mitch");
+					});
+				});
+		});
+
+		it("status: 404, should return not found for a non-existent topic", () => {
+			return request(app)
+				.get("/api/articles?topic=non_existent_topic")
+				.expect(404)
+				.then(({ body }) => {
+					expect(body.message).toBe("no article found under non_existent_topic");
+				});
+		});
```

Comment:
Ideally, we want our error messages to be generic. Being overly specific error messages gives others an idea of how to access specific information in your database 🔍